src/components/App/App.vue

```diff
@@ -43,9 +49,46 @@ export default {
       })
     },
     retrieveLocale (location) {
-      fetchAPI.getLocale(location.city, location.state).then(data => {
-        this.locations.push(data.data)
+      fetchAPI.getLocale(location.city, location.state).then(value => {
+        value.data.id = Date.now()
+        this.checkExistingLocations(value.data)
+        this.updateLocalStorage()
+      })
+    },
+    retrieveCurrent () {
+      fetchAPI.getCurrent().then(location => {
+        location.data.id = Date.now()
+        this.checkExistingLocations(location.data)
+        this.updateLocalStorage()
+      })
+    },
+    removeLocation (id) {
+      const thisLocation = this.locations.find(location => {
+        return location.id === parseInt(id)
       })
+      const thisLocationIndex = this.locations.indexOf(thisLocation)
+      this.locations.splice(thisLocationIndex, 1)
+      this.updateLocalStorage()
+    },
+    checkExistingLocations (newLocation) {
+      const foundLocation = this.locations.find(location => {
+        return location.city === newLocation.city && location.state === newLocation.state
+      })
+      if (foundLocation) {
+        const foundLocationIndex = this.locations.indexOf(foundLocation)
+        this.locations[foundLocationIndex] = newLocation
+      } else {
+        this.locations.push(newLocation)
+      }
+    },
+    retrieveLocalStorage () {
```

Comment:
Glad we added this! Wonder if we should also add some copy that lets people know their locations will be saved for then they come back?